pkg/kube/util_test.go

```diff
@@ -329,6 +329,48 @@ func TestGenericReplacement_missingValue(t *testing.T) {
 	assertFailedReplacement(&dummyResource, &expected, t)
 }
 
+func TestGenericReplacement_allowEmpty(t *testing.T) {
+	dummyResource := Resource{
+		TemplateData: map[string]interface{}{
+			"namespace": "default",
+			"annotations": map[string]interface{}{
+				"avp.kubernetes.io/allow-empty": "true",
+			},
+			"spec": map[string]interface{}{
+				"replicas": "<replicas>",
+			},
+		},
+		Data: map[string]interface{}{
+			"namespace": "default",
+		},
+		Annotations: map[string]string{
+			"avp.kubernetes.io/allow-empty": "true",
+		},
+	}
+
+	replaceInner(&dummyResource, &dummyResource.TemplateData, genericReplacement)
+
+	expected := Resource{
+		TemplateData: map[string]interface{}{
+			"namespace": "default",
+			"annotations": map[string]interface{}{
+				"avp.kubernetes.io/allow-empty": "true",
+			},
+			"spec": map[string]interface{}{
+				"replicas": "<replicas>",
+			},
+		},
+		Data: map[string]interface{}{
+			"namespace": "default",
+		},
+		Annotations: map[string]string{
+			"avp.kubernetes.io/allow-empty": "true",
+		},
+	}
+
+	assertFailedReplacement(&dummyResource, &expected, t)
```

Comment:
Why are we asserting failed here? If I read this right, the presence of the new annotation makes it so that `replacementErrors` is left empty if a placeholder can't be replaced. With that and the expected and actual data matching, this test seems wrong to me.

Reply:
Ah I just copied a test above, I think both assertions would work cause we're just trying to assert that there are no errors. I can push a change making it `assertSuccessfulReplacement` if thats clearer? Or should I make a new assertion?

Reply:
> I can push a change making it assertSuccessfulReplacement

Yes, this would be better and looks like you did that. Thanks!